working-groups/meta/rfc-drafts/compiler-team-contributors.md

```diff
@@ -0,0 +1,244 @@
+have r+ privileges.
+
+Often, it makes sense to be added to the queue *even if* you are not
+that familiar with the compiler. This is because it lets you do
+initial reviews of PRs, thus gaining experience with lots of parts of
+the compiler. If you don't feel like you fully understood the PR, then
+-- after your initial review -- you can then re-assign the PR to
+someone more senior.
+
+### rust-lang org membership
+
+There are a number of things that you can't do in GitHub unless you
+are a member of the GitHub organization. Typically, one becomes a
+member of the organization by being added to a team, and these teams
+in turn are granted privileges to repositories in the organization.
+Most notably:
+
+- you cannot be assigned to issues unless you have at least *read* access to a repository;
+- you cannot modify labels without *write* access (I think);
+- you cannot be a member of a *team*, which means you cannot be addressed via some
+  alias like `@rust-lang/compiler-team`;
+- you do not get the little "member" badge appearing next to your name when you comment.
+
+The last point is potentially important: by being made a member of the
+org, you are to some extent representing that org, as you are
+visibility identified as a member.  These can be important in terms of
+the code of conduct, as we wish for representatives of rust-lang to
+take extra care in their public interaction. In particular, this
+implies we might not want to allow **anyone** to make themselves a
+member of the org.
+
+### triagebot
+
+The triagebot is an "upcoming" piece of infrastructure that should allow any GitHub user
+to make some changes to issues on rust-lang repositories. In particular, one would be
+able to instruct the triagebot to do the following:
+
+- adjust labels on issues
+- assign oneself to the issue
+
+Because the triagebot can be used by anyone, and not just org members, assigning
+works as follows:
+
+- the issue is *officially* assigned to the triagebot (as far as
+  Github is concerned, that is)
+- the issue header is edited to indicate that it is assigned to the
+  user in question
+
+This is a bit less good than being assigned to the issue as an org
+member, since it means that your username and picture do not appear
+next to the issue, but it's still pretty decent and should suffice for
+most purposes.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+## The path to membership
+
+People will typically start as a **working group participant,** which
+is basically somebody who has come to work on something for the first
+time. They don't know much about the compiler yet and have no
+particular privileges. They are assigned to issues using the triagebot
+and (typically) work with a mentor or mentoring instructions.
+  
+Once a working group participant has been contributing regularly for
+some time, they can be promoted to the level of a **compiler team
```

Comment:
I think I'd like to see this part expanded on. I don't think that we should specify an exact timeframe or number of PRs or anything like that - I think we do want to be vague, but something that sets expectations would probably be useful - different contributors will have different ideas of what "regularly for some time" means.

Reply:
I see your point @davidtwco, unsure how this can be expanded though. I think it's usual on OSS that after some time someone or a team makes a call and the contributor is in some way "promoted".

Reply:
I agree, I'm not sure what it would look like either.

Reply:
I think I agree with both of you. I'd love to write more, but I'm not *quite* sure what to write. Partly because I think it's sort of hard to judge what is the right point. But I think there would be value in taking a stab at it, because I could imagine many people having very different opinions.

I think my feeling would be something like this:

- if the person has completed 2 or 3 independent, non-trivial "projects" 
- the person participates regularly in the working group, either via meetings, being active on Zulip, or whatever

Basically, have they shown 'staying power'

Reply:
I've added language here